### PR TITLE
Always run detection of libc on native target

### LINKED ISSUE #4772: support detecting MSVC v142 - VS 2019 C++ build tools v14.2
On Windows, when adding `.linkSystemLibrary("c");` to my `build.zig` I get the following errors. 
```
lld: error: could not open 'libcmtd.lib': no such file or directory
lld: error: could not open 'libvcruntimed.lib': no such file or directory
lld: error: could not open 'legacy_stdio_definitions.lib': no such file or directory
```

I assume this is because I updated my visual studio installation from 16.4.something (I think it was .4) to 16.5.0. I also updated my zig before trying to compile my project where I link against C libraries. I tried downgrading to what I think was my previous zig version, but it didn't make any difference.

I'm currently at zig 0.5.0+555a2c032

Any ideas on what might be wrong or how to fix it are welcome. 

I do have the libraries in: `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\lib\x64`

### PR DESCRIPTION
**Pull request by mlarouche**
Always run detection of libc on native target

Fixes #4772

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1361,10 +1361,6 @@ static int main0(int argc, char **argv) {
             fprintf(stderr, "unable to find native libc file: %s\n", err_str(err));
             return main_exit(root_progress_node, EXIT_FAILURE);
         }
-        if ((err = stage2_libc_render(&libc, stdout))) {
-            fprintf(stderr, "unable to print libc file: %s\n", err_str(err));
-            return main_exit(root_progress_node, EXIT_FAILURE);
-        }
```
Comment (andrewrk) [inline src/main.cpp:1367]: This part should stay. This is when you call `zig libc` and it prints the libc file, which you would then edit (or not) and pass to `--libc`.

Comment (mlarouche) [reply src/main.cpp]: Updated my PR

**Comment by andrewrk** [comment]
Thanks! cc @mikdusan

**Comment by mikdusan** [comment]
I thought we wanted to avoid spawning `cc -E -Wp,-v -xc /dev/null` on every `zig build-<whatever>` command?

**Comment by andrewrk** [comment]
That would be nice but it's better than the alternative of dealing with stale cache bugs.

Here's a proposal attempting to address both use cases:

 * Revive the code that looks for a project-local and global `native_libc.txt` file.
 * However, do not create this file automatically. Taking advantage of this feature would require the user to manually do `zig libc >$PROJECTDIR/zig-cache/native_libc.txt` or `zig libc > $the_global_dir/native_libc.txt`

It could potentially also work with an environment variable `ZIG_NATIVE_LIBC` which would be manually set by the user.

I propose this because it has the properties that, by default, it always works with no cache issues. But if a user wants to take on additional usage complexity to gain an efficiency improvement, they are able to do so.

**Comment by mikdusan** [comment]
I'm ok with the patch as-is... just wanted to explicitly mention the cost in case it was missed.

We can make sure there is a more optimal solution when std.build matures. In the back of my mind I've been thinking we punt caching of things like this to std.build, and then it can pass the --libc option to zig as appropriate. To elaborate a little, we don't have an analogue to  "cmake" setup/configuration of a project yet but we'll eventually need a solve, and wherever that build configuration is cached is naturally a place `native_libc.txt` .

**Pull request merged**